From e4006e6066c015d9ed55befa9b98247fbdcafd7d Mon Sep 17 00:00:00 2001
From: Erik Larsson <who+github@cnackers.org>
Date: Mon, 26 Jun 2023 12:15:41 +0200
Subject: [PATCH] test: add check for renamed cryptography types

Some types have changed their names in newer cryptography release, so add them to the tests

Signed-off-by: Erik Larsson <who+github@cnackers.org>
---
 test/test_crypto.py | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/test/test_crypto.py b/test/test_crypto.py
index 92cda00..7d7466e 100644
--- a/test/test_crypto.py
+++ b/test/test_crypto.py
@@ -596,11 +596,23 @@ class CryptoTest(TSS2_EsapiTest):
 
         with self.assertRaises(ValueError) as e:
             TPMT_SENSITIVE.from_pem(der)
-        self.assertEqual(str(e.exception), "unsupported key type: _DSAPrivateKey")
+        self.assertIn(
+            str(e.exception),
+            (
+                "unsupported key type: _DSAPrivateKey",
+                "unsupported key type: DSAPrivateKey",
+            ),
+        )
 
         with self.assertRaises(ValueError) as e:
             TPMT_PUBLIC.from_pem(dsa_public_key)
-        self.assertEqual(str(e.exception), "unsupported key type: _DSAPublicKey")
+        self.assertIn(
+            str(e.exception),
+            (
+                "unsupported key type: _DSAPublicKey",
+                "unsupported key type: DSAPublicKey",
+            ),
+        )
 
     def test_from_pem_with_symmetric(self):
         sym = TPMT_SYM_DEF_OBJECT(algorithm=TPM2_ALG.AES)
-- 
2.41.0

